# Testing 3V Analog Voltmeter

In this project we will test the $3\mathrm{V}$ [analog voltmeter](https://www.adafruit.com/product/3989) by connecting it in parallel with a small resistor that is in series with a large resistor, a $1.85\mathrm{V}$ (typical) [LED](https://cdn-shop.adafruit.com/datasheets/WP7113SRD-D.pdf), and a $9\mathrm{V}$ DC voltage source.  The aforementioned voltage source is actually a [DC barrel jack](https://www.adafruit.com/product/373) connected to a $9\mathrm{V}$ Duracell battery.  

Note that the LED is to provide  another indication that the circuit is closed.

### Ignoring the Barrel Jack
Assuming the DC barrel jack's characteristics have insignificant effects on the rest of the circuit, and thus ignoring it all together, the circuit is of the form:  $9\mathrm{V}--1.85\mathrm{V}--R_1--R_2//R_\mathrm{V}$  where $R_1>>R_2$ and '$R_\mathrm{V}$' denotes the resistance of the voltmeter.

Via KVL and Ohm's Law,  

\begin{equation}\tag{1}
    7.15\mathrm{V} = R_1I_1 + R_2I_2
\end{equation}    

\begin{equation}\tag{2}
    R_2I_2 = R_\mathrm{V}I_\mathrm{V}
\end{equation}    

Via KCL,  

\begin{equation}\tag{3}
    I_1 = I_2 + I_\mathrm{V}
\end{equation}    

Via equations (3) and (1) and substitution,  

\begin{equation}\tag{4}
    7.15\mathrm{V} = R_1(I_2 + I_\mathrm{V}) +R_2I_2
\end{equation}    

Via equations (4) and (2),  

$$ 7.15\mathrm{V} = R_1\left(I_2 + \frac{R_2I_2}{R_\mathrm{V}}\right) + R_2I_2$$

$$ 7.15\mathrm{V} = R_1I_2\left(1 + \frac{R_2}{R_\mathrm{V}}\right) + R_2I_2= R_1I_2\,\frac{R_\mathrm{V} + R_2}{R_\mathrm{V}} + R_2I_2= I_2\left(R_1\,\frac{R_\mathrm{V}+R_2}{R_\mathrm{V}} + R_2\right)$$

$$ 7.15\mathrm{V} = I_2\,\frac{R_1(R_\mathrm{V}+R_2) + R_\mathrm{V}R_2}{R_\mathrm{V}}$$

\begin{equation}\tag{5}
    I_2 = \frac{7.15\mathrm{V}R_\mathrm{V}}{R_1(R_\mathrm{V}+R_2) + R_\mathrm{V}R_2}
\end{equation}    

Via equations (2) and (5) and Ohm's Law,  

\begin{equation}\tag{6}
    U_\mathrm{V} = 7.15\mathrm{V}\,\frac{R_2R_\mathrm{V}}{R_1(R_\mathrm{V}+R_2) + R_\mathrm{V}R_2}
\end{equation}    

In [2]:
def U_V_gen(R_1, R_2):
    return lambda R_V: 9*R_2*R_V/(R_1*(R_V+R_2) + R_V*R_2)

In [3]:
U_V = U_V_gen(1e3, 330)

In [4]:
U_V(1e6)

2.2325287710568054

Via equations (1) and (5) and Ohm's Law,  

$$ U_1 = 7.15\mathrm{V}\,\left(1 - \frac{R_2R_\mathrm{V}}{R_1(R_\mathrm{V}+R_2) + R_\mathrm{V}R_2}\right)$$

\begin{equation}\tag{9}
    U_1 = 7.15\mathrm{V}\,\frac{R_1(R_\mathrm{V}+R_2)}{R_1(R_\mathrm{V}+R_2) + R_\mathrm{V}R_2}
\end{equation}    

In [5]:
def U_1_gen(R_1, R_2):
    return lambda R_V: 9*R_1*(R_V+R_2)/(R_1*(R_V+R_2) + R_V*R_2)

In [6]:
U_1 = U_1_gen(1e3, 330)

In [7]:
U_1(1e5)

6.772444311107778

Assuming $R_2 = 1\mathrm{k}\Omega$ and $R_1=10\mathrm{k}\Omega$, but $R_\mathrm{V}$ is unknown, we can find $R_\mathrm{V}$ by taking a voltmeter indication ($U_\mathrm{V}$) and using equation (6) to solve for $R_\mathrm{V}$.  Via equation (6),  

$$ U_\mathrm{V}\big(R_1(R_\mathrm{V}+R_2) + R_\mathrm{V}R_2 \big) = 7.15\mathrm{V}R_2R_\mathrm{V}$$

$$  U_\mathrm{V}R_1R_2  = 7.15\mathrm{V}R_2R_\mathrm{V} -U_\mathrm{V}R_1R_\mathrm{V} - U_\mathrm{V}R_\mathrm{V}R_2$$

$$ R_\mathrm{V}(7.15\mathrm{V}R_2 - U_\mathrm{V}R_1 - U_\mathrm{V}R_2) = U_\mathrm{V}R_1R_2$$

Finally,
\begin{equation}\tag{7}
    R_\mathrm{V} = \frac{U_\mathrm{V}R_1R_2}{7.15\mathrm{V}R_2 - U_\mathrm{V}R_1 - U_\mathrm{V}R_2}
\end{equation}    

In [8]:
def R_V(U_V, R_1, R_2, E = 9):
    return U_V*R_1*R_2/(E*R_2 - U_V*R_1-U_V*R_2)

In [9]:
R_V(0.5, 1e4, 1e3, 9-1.85)

3030.3030303030305

The **voltmeter indication**, denoted as '$U_\mathrm{V}$', was roughly $0.5\mathrm{V}$ with $R_1\approx 10\mathrm{k}\Omega$ and $R_2\approx 1\mathrm{k}\Omega$.  Thus, $R_\mathrm{V}\approxeq 3030.30\Omega$.

Note that $U_\mathrm{V} \le 3\mathrm{V}$.  Via the aforementioned inequality and equation (6),  

$$ 3R_2R_\mathrm{V} \le  R_1(R_\mathrm{V}+R_2) + R_\mathrm{V}R_2$$

$$ R_\mathrm{V}(3R_2 - R_1 - R_2)\le R_1R_2$$

\begin{equation}\tag{8}
    R_\mathrm{V} \ge \frac{R_1R_2}{\vert 2R_2-R_1\vert} \approxeq \frac{10\mathrm{M}\Omega^2}{8\mathrm{k}\Omega} = 1.25\mathrm{k}\Omega
\end{equation}    